addons/stock/models/stock_move.py

```diff
@@ -505,31 +504,17 @@ def _do_unreserve(self):
         return True
 
     def _push_apply(self):
-        # TDE CLEANME: I am quite sure I already saw this code somewhere ... in routing ??
-        Push = self.env['stock.location.path']
         for move in self:
             # if the move is already chained, there is no need to check push rules
-            if move.move_dest_ids:
+            if move.move_dest_ids or move.procure_method == 'make_to_order':
```

Comment:
seems weird
the last move of a buy + pick pack ship is mto and we should be apply to call _push_apply